### ISSUE TITLE
Datefy message.date

### ISSUE THREAD

**Issue by leandrotoledo** [leandrotoledo/python-telegram-bot#35]
Right now it returns a string as Unix date format

**Comment by JokerQyou** [comment]
A datetime.datetime object representing the UTC time would be great.

**Comment by leandrotoledo** [comment]
Oops, I forgot an ``import pdb`` on tests, resolved on fda1843593c22210a196ae1a17dc8acc22270a05

**Comment by The-Fonz** [comment]
Currently, due to the use of `.fromtimestamp(...)` instead of `.utcfromtimestamp(...)`, the *message.date* and other datetime attributes are naive (non-timezone-aware) local times. To convert these back to UTC time requires knowledge of the local timezone (needs the *tzlocal* package) and conversion, something like this:

```
# pip install pytz tzlocal
import pytz
import tzlocal

def localtime_to_utc(dt):
    "Converts naive datetime.datetime object from local timezone to UTC"
    # First find out local timezone
    localzone = tzlocal.get_localzone()
    # Now add local timezone information
    dt = localzone.localize(dt)
    # Finally convert to UTC time, using .isoformat() it will show in UTC
    return dt.astimezone(pytz.utc)
```

It seems much better to me to use `.utcfromtimestamp()` so that the naive datetime.datetime object is already in UTC. Converting to a specific timezone is much easier than converting it back and having to guess the timezone.

But I'm not sure if this would break any applications. Something that should be non-breaking is to add timezone information using the *tzlocal* package as shown above, this would however add an extra dependency. A third option would be to add a setting somewhere in the updater function call or elsewhere that passes the times as their original Unix time float. Or extra properties could be added to the *Message* object that store the time in UTC.

What do you guys think?
